modules/schematics/src/create-action-migration/index.ts

```diff
@@ -0,0 +1,206 @@
+import * as ts from 'typescript';
+import { Path, tags } from '@angular-devkit/core';
+import { camelize } from '@angular-devkit/core/src/utils/strings';
+import { Tree, Rule, chain } from '@angular-devkit/schematics';
+import {
+  InsertChange,
+  RemoveChange,
+  ReplaceChange,
+  replaceImport,
+  commitChanges,
+  Change,
+} from '@ngrx/schematics/schematics-core';
+
+export function migrateToCreators(): Rule {
+  return (host: Tree) =>
+    host.visit(path => {
+      if (!path.endsWith('.ts') || path.includes('node_modules')) {
+        return;
+      }
+
+      const sourceFile = ts.createSourceFile(
+        path,
+        host.read(path)!.toString(),
+        ts.ScriptTarget.Latest
+      );
+
+      if (sourceFile.isDeclarationFile || !isActionImported(sourceFile)) {
+        return;
+      }
+
+      const actionClasses = getActionClasses(sourceFile);
+      const actionTypeVariables = sourceFile.statements.filter(
+        ts.isVariableStatement
+      );
+
+      let importProps = false;
+
+      let changes = (actionClasses
+        .map(actionClass => {
+          let typeProperty = actionClass.members
+            .filter(ts.isPropertyDeclaration)
+            .filter(member => (member.name as ts.Identifier).text === 'type')
+            .map(member => (member.initializer as ts.Identifier).text)[0];
+
+          if (!typeProperty) {
+            return undefined;
+          }
+
+          const typeVariable = actionTypeVariables.filter(
```

Comment:
I would rather see `actionTypeVariables` as a lookup, the key can be the variable name and the value can be the string value. That way we prevent unnecessary lookups here and would it also make more readable imho.